presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -474,7 +474,14 @@ public StreamProperties visitTopN(TopNNode node, List<StreamProperties> inputPro
         @Override
         public StreamProperties visitSort(SortNode node, List<StreamProperties> inputProperties)
         {
-            return StreamProperties.ordered();
+            StreamProperties sourceProperties = Iterables.getOnlyElement(inputProperties);
+            if (sourceProperties.isSingleStream()) {
+                // stream is only sorted if sort operator is executed without parallelism
+                return StreamProperties.ordered();
+            }
+            else {
+                return sourceProperties;
```

Comment:
Are all previous properties retained, always?

Reply:
I think so.

On the side node I see that: `StreamProperties#otherActualProperties` will contain `SortingProperty` for the node itself even if the parallel stream is not ordered, but I don't think that is a problem